src/configure/auto_outlook.rs

```diff
@@ -1,122 +1,178 @@
 //! Outlook's Autodiscover
 
-use quick_xml::events::BytesEnd;
+use quick_xml::events::Event;
+
+use std::io::BufRead;
 
 use crate::context::Context;
 use crate::login_param::LoginParam;
-use crate::provider::Socket;
+use crate::provider::{Protocol, Socket};
 
 use super::read_url::read_url;
-use super::Error;
-
-struct OutlookAutodiscover {
-    pub out: LoginParam,
-    pub out_imap_set: bool,
-    pub out_smtp_set: bool,
-    pub config_type: Option<String>,
-    pub config_server: String,
-    pub config_port: u16,
-    pub config_ssl: String,
-    pub config_redirecturl: Option<String>,
+use super::{Error, ServerParams};
+
+#[derive(Debug)]
+struct ProtocolTag {
```

Comment:
Any chance we can get some more doc strings in here? I know there weren't any around before but would be quite helpful for the future.

Reply:
Added docs to the structure and all its members, and extended module header a bit.